src/main/java/randoop/main/GenInputsAbstract.java

```diff
@@ -699,6 +699,19 @@ public enum BloodhoundCoverageUpdateMode {
   @Option("Maximum length of Strings in generated tests")
   public static int string_maxlen = 1000;
 
+  /**
+   * The "Detective" technique from the GRT paper attempts to construct missing inputs on demand. By
+   * default, when calling a method, Randoop uses as arguments whatever values Randoop has already
+   * generated. This may prevent Randoop from calling a method, if Randoop has not yet generated any
+   * values of the appropriate type. When Detective is enabled, Randoop immediately attempts to
+   * construct inputs for the method under test.
+   *
+   * <p>The default value is {@code false}.
+   */
+  @Unpublicized
+  @Option("Construct method inputs on demand, if no value exists yet of the given type")
+  public static boolean detective = false;
```

Comment:
As discussed elsewhere, please rename.

Reply:
I'm a little out-of-idea on finding a good name for it in command line option. I considered using "demand-driven-input-creation" (directly after the new name) but it might not be a good idea for some reasons. In my opinion, both "detective" and "demand-driven-input-creation" can be equally confusing and not very informative to users without reading the randoop manual. If they do read, the name itself will no longer help them understand the component's functionality because documentation is supposed to provide more information than just the name, and at that point, it is possible that using "detective" will make the randoop command cleaner. Ideally, we want it to be a good reminder or hint to users about what "detective" is doing without adding more mess to Randoop's already difficult and confusing usage, and I believe the benefits of the name "detective" outweigh the drawbacks of changing it to something much longer. It is also consistent with other GRT components that are currently in use, such as "bloodhound". Do you have any suggestions for a better name or reasons why my concerns are unlikely to be a big deal? Thanks!